native/packages/react-native-bpk-component-text/src/common-types.js

```diff
@@ -20,70 +20,110 @@
 
 import { type Node } from 'react';
 import PropTypes from 'prop-types';
-import { Platform, StyleSheet, type StyleObj } from 'react-native';
+import { StyleSheet, type StyleObj } from 'react-native';
 
-export const TEXT_STYLES = ['xs', 'sm', 'base', 'lg', 'xl', 'xxl'];
-export type TextStyle = 'xs' | 'sm' | 'base' | 'lg' | 'xl' | 'xxl';
+export const TEXT_STYLES = [
+  'caps',
+  'xs',
+  'sm',
+  'base',
+  'lg',
+  'xl',
+  'xxl',
+  'xxxl',
+];
+export type TextStyle =
+  | 'caps'
+  | 'xs'
+  | 'sm'
+  | 'base'
+  | 'lg'
+  | 'xl'
+  | 'xxl'
+  | 'xxxl';
+
+export const WEIGHT_STYLES = {
+  regular: 'regular',
+  emphasized: 'emphasized',
+  heavy: 'heavy',
+};
+export type Weight = $Keys<typeof WEIGHT_STYLES>;
 
 export type Props = {
   children: Node,
-  textStyle: TextStyle,
-  emphasize: boolean,
+  emphasize: ?boolean,
   style: ?StyleObj,
+  textStyle: TextStyle,
+  weight: Weight,
 };
 
-export const emphasizePropType = (
+export const stylePropType = (
   props: Props,
   propName: string,
   componentName: string,
 ) => {
-  const value = props[propName];
-  if (typeof value !== 'boolean') {
-    return new Error(
-      `Invalid prop \`${propName}\` of type \`${typeof value}\` supplied to \`${componentName}\`, expected \`boolean\`.`,
-    ); // eslint-disable-line max-len
-  }
+  const value = StyleSheet.flatten(props[propName]);
 
-  const enabled = !!value;
+  if (value === undefined) return false;
 
-  if (Platform.OS === 'ios' && (enabled && props.textStyle === 'xxl')) {
+  if (value.fontWeight) {
     return new Error(
-      `Invalid prop \`${propName}\` of value \`${value}\` supplied to \`${componentName}\`. On iOS, \`textStyle\` value of \`xxl\` cannot be emphasized.`,
+      `Invalid prop \`${propName}\` with \`fontWeight\` value \`${
+        value.fontWeight
+      }\` supplied to \`${componentName}\`. Use \`emphasize\` prop instead.`,
     ); // eslint-disable-line max-len
   }
 
   return false;
 };
 
-export const stylePropType = (
+export const isWeightValid = (weight, textStyle) =>
+  weight !== WEIGHT_STYLES.heavy ||
+  (weight === WEIGHT_STYLES.heavy && textStyle.includes('xl'));
```

Comment:
Can we comment the magic going on here with `includes` and `xl`?

Reply:
Commit `discussion_r228179477`